paperqa/docs.py

```diff
@@ -80,6 +80,12 @@ class Docs(BaseModel):
         default=PAPERQA_DIR, description="Path to save index", validate_default=True
     )
     deleted_dockeys: set[DocKey] = Field(default_factory=set)
+    partitioning_fn: Callable[[Doc], int] | None = Field(
```

Comment:
I thought that too-- but I think the serialization issue seems worse there since that's the most frequently serialized object we use. The `Settings` objects are often initialized from a validated JSON (which this attribute wouldn't work with) and the `Docs` are initialized via code (or a deserialization from a pickle), but `Docs` objects already aren't JSON compatible because of the VectorStore. 

What do you think?

Also could you elaborate on "better mapping between str and functions"? 

In practice, I'm modifying this attribute dynamically upon env initialization if a particular tool is selected, so we could tack it on to a settings object just while it's in the lifecycle of the environment, then remove it, or just exclude from serialization in the same way I'm doing here.

Reply:
Actually thought more about this — I think I’ll just add as an arg to all the requisite functions— then we skip this issue entirely. I’ll need to add the funcs to the tool initializations, which I’ll think through more.

Reply:
Ok all set now! No more stateful stuff.